app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,64 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = [];
+
+    $('#hosts-table tbody tr').each( function (index, item) {
+      res.push($(item).data("host-id"));
+    });
+
+    return res;
+  }
+
+  function findSpinner(id) {
+    return $('#hosts-table tbody').find("tr[data-host-id='" + id + "']").find(".spinner");
+  }
+
+  function updatePowerStatus(response) {
+    var spinner = null;
+    hostIds().forEach(function (item) {
+      spinner = findSpinner(item);
+      if ((typeof response !== "string") && response[item]) {
+        hostPowerStatus(spinner, response[item]);
+      } else {
+        hostPowerStatus(spinner, false);
+      }
+    });
+  }
+
+  function handleError(response) {
+    hostIds().forEach(function (item) {
+      hostPowerStatus(findSpinner(item), false);
+    });
+  }
+
+  function hostPowerStatus(spinner, power) {
+    if (power) {
+      $(spinner).nextAll(".label-success").removeClass("hidden");
```

Comment:
`spinner` is already a jquery object in this function, no need to use `$(spinner)`